fineract-provider/src/main/java/org/apache/fineract/infrastructure/core/domain/AuditorAwareImpl.java

```diff
@@ -35,7 +35,7 @@ public Optional<Long> getCurrentAuditor() {
         final SecurityContext securityContext = SecurityContextHolder.getContext();
         if (securityContext != null) {
             final Authentication authentication = securityContext.getAuthentication();
-            if (authentication != null) {
+            if (authentication != null && authentication.getPrincipal() instanceof AppUser) {
```

Comment:
I dont like this... from audit point of view, falling back to super user in this case is incorrect.